cmd/tiller-proxy/main.go

```diff
@@ -99,6 +100,37 @@ func main() {
 		log.Fatalf("Unable to create a kubernetes client: %v", err)
 	}
 
+	caCertPool := x509.NewCertPool()
+	systemCACert, err := ioutil.ReadFile("/etc/ssl/certs/ca-certificates.crt")
+	if err != nil {
+		log.Fatalf("Unable to read system CA certs: %v", err)
+	}
+	caCertPool.AppendCertsFromPEM(systemCACert)
+
+	if len(registryCaCerts) > 0 {
+		namespace := os.Getenv("POD_NAMESPACE")
+		for _, registryCaCert := range registryCaCerts {
+			caCertSecret, err := kubeClient.CoreV1().Secrets(namespace).Get(registryCaCert, metav1.GetOptions{})
```

Comment:
I am not convinced that referencing from here k8s is a good idea. Why isn't it loading it from disk or env?